utils/unminify.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2017 The AMP Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Convert's stacktrace line, column number and file references from minified
+ * to unminified. Caches requests for and source maps and once obtained.
+ * }
+ */
+
+const sourceMap = require('source-map');
+const Request = require('./request');
+const lineColumnNumbersRegex = '([^ \\n]+):(\\d+):(\\d+)';
+/** @type {!sourceMap<url, sourceMap>}*/
+const sourceMapConsumerCache = new Map();
+/** @type {!request<url, Promise>}*/
+const requestCache = new Map();
+
+/**
+ * @param {string} stackTraceLine
+ * @param {Object} sourceMapConsumer
+ * @return {string} Stack trace line with column, line number and file name
+ * references unminified.
+ */
+function unminifyLine(stackTraceLine, sourceMapConsumer) {
+  const chromeStackTraceRegex = new RegExp(
+      `^\\s*at (.+ )?(?:((${lineColumnNumbersRegex}))|` +
+      `\\((${lineColumnNumbersRegex}\\)))$`, 'gm');
+  [_, _, _, _, sourceUrl, lineNumber, columnNumber] =
```

Comment:
You don't have to use `_` for ignored values.

`[,,,, sourceUrl, lineNumber, columnNumber]`